pom.xml

```diff
@@ -229,6 +235,17 @@ knowledge of the CeCILL-B license and that you accept its terms.
 
     <build>
         <plugins>
+            <plugin>
+                <!-- needed for Spring 6.1 -->
```

Comment:
Without javac `-parameters` flag, Spring reflection fails with `IllegalArgumentException` at runtime, when authenticating a request with an "apikey: ..."  header, such as a `GET /rest/executions`. Here's a more detailed trace below.

I'm still not 100% clear on whether this could be solved with some extra annotations at the right place : I tried a few things with no success, and ended up adding this parameter, which is consistent with the recommendations in https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-6.1-Release-Notes#parameter-name-retention .

`
2024-11-05 09:42:38,672 |ERROR |http-nio-8080-exec-6 | f.i.c.v.a.c.RestExceptionHandler:handleAllException:142 - Unexpected exception catched
 Caller+0        at fr.insalyon.creatis.vip.api.controller.RestExceptionHandler.handleAllException(RestExceptionHandler.java:142)
Caller+1         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caller+2         at java.base/java.lang.reflect.Method.invoke(Method.java:580)
Caller+3         at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:255)
Caller+4         at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:188)
Caller+5         at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:118)
Caller+6         at org.springframework.web.servlet.mvc.method.annotation.ExceptionHandlerExceptionResolver.doResolveHandlerMethodException(ExceptionHandlerExceptionResolver.java:432)
Caller+7         at org.springframework.web.servlet.handler.AbstractHandlerMethodExceptionResolver.doResolveException(AbstractHandlerMethodExceptionResolver.java:74)
java.lang.IllegalArgumentException: Name for argument of type [java.lang.String] not specified, and parameter name information not available via reflection. Ensure that the compiler uses the '-parameters' flag.
        at org.springframework.web.method.annotation.AbstractNamedValueMethodArgumentResolver.updateNamedValueInfo(AbstractNamedValueMethodArgumentResolver.java:186) ~[spring-web-6.1.14.jar:6.1.14]
        at org.springframework.web.method.annotation.AbstractNamedValueMethodArgumentResolver.getNamedValueInfo(AbstractNamedValueMethodArgumentResolver.java:161) ~[spring-web-6.1.14.jar:6.1.14]
...
`

Reply:
Indeed, this is necessary, or we have to update all our `@PathVariable` and `@RequestVariable` with  `@PathVariable("executionId") for instance. This would be doable but would make the method signatures longer and harder to understand.
As the -parameters option seems recomended by Spring, let's keep it that way.